src/Domain/FunctionsOfTime/OptionTags.hpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "Options/Auto.hpp"
+#include "Options/Options.hpp"
+
+namespace domain::FunctionsOfTime::OptionTags {
+/*!
+ * \ingroup OptionGroupsGroup
+ * \brief Groups options for reading in FunctionOfTime data from SpEC
+ */
+struct CubicFunctionOfTimeOverride {
+  static constexpr Options::String help{
+      "Options for importing FunctionOfTimes from SpEC"};
+};
+
+/*!
+ * \brief Path to an H5 file containing SpEC FunctionOfTime data
+ */
+struct FunctionOfTimeFile {
+  using type = Options::Auto<std::string>;
+  static constexpr Options::String help{
+      "Path to an H5 file containing SpEC FunctionOfTime data"};
+  using group = CubicFunctionOfTimeOverride;
+  static type default_value() noexcept { return {}; }
```

Comment:
Without the default the option reads `FunctionOfTimeFile: Auto` but it means "don't read in a file". I'm not arguing for keeping the default, but pointing out this issue with our option parsing. Possible resolutions (out of scope for this PR):
- Allow customizing the `Options::Auto::name()`. I couldn't think of a way to do that and thus didn't brought it up when reviewing #2512, because it's hard to use strings as template parameters.
- Add `Options::None` or just make `std::optional` option-creatable. Less flexible, but may cover the cases that `Auto` doesn't.
@wthrowe could you give your opinion on this?

Reply:
I added the default value so that this change doesn't break everyone's input files, requiring them to add an option that is really only needed for GH.

Reply:
> Is there a better way to avoid requiring every input file to give a FunctionOfTimeFile: and FunctionofTimeNameMap: option, when it's not really appropriate except for GH when importing from SpEC?

The current plan is to make all options required.  Any discussion of that should occur on #2522.  Unless you want to object to the current plan, don't add a default because it will just mean I have to go through the work of removing it when I rebase #2522.

In this case, if I'm reading your code correctly you know at compile time whether you want these options, so can you only request them if you need them?

Reply:
Actually you already have that compile-time option implemented in your `InitialFunctionsOfTime` tag, so all you need to do is toggle it in the GH exec.

Reply:
@wthrowe do you have an opinion on my `Auto`/`None` comment above?

Reply:
I chose not to make std::optional parsable before because there was no obvious representation for the empty state.

How about
```c++
enum class AutoLabel { Auto, None };
template <typename T, AutoLabel Label = AutoLabel::Auto>
class Auto { ... };
```

Or, since these cases are usually parsing strings, you could special-case the value "None" yourself in the constructor.

Reply:
The enum-approach sounds good. The constructor-approach I don't quite understand. Which constructor are you suggesting to special-case?

Reply:
@wthrowe @nilsleiffischer I'm just going to remove the default options and ensure that my new options are only requested when needed, using my compile-time flag. Am I correct that this discussion about using an enum for parsing std::optional types is something beyond the scope of this PR?

Reply:
@nilsleiffischer Sorry, I meant in `create_from_options`, not a constructor.  (I'm more used to the class-creation code.)